### ISSUE TITLE
macOS support

### ISSUE THREAD

**Issue by ProjectInfinity** [JetpackDuba/Gitnuro#5]
Hi, I built this project successfully on a Apple Silicone mac.

In doing so I discovered that the version string breaks the native macOS builder during compose's task. 
The error in particular was:
`Bundler Mac Application Image skipped because of a configuration problem: The first number in an app-version cannot be zero or negative. 
Advice to fix: Set a compatible 'app-version' value. Valid versions are one to three integers separated by dots.`

The fix for me was to open build.gradle.kts and changing line 14 to say
`val projectVersion = "1.0.0"`

**Comment by JetpackDuba** [comment]
Hey!

Yes, I am aware of the issue. The same problem shows up when building `.deb` for Debian based linux distros.

I'm probably not going to change it for now as in july I'll release the 1.0.0 version. Basically because I need some free time to finish a few features, setup an env to test the MacOS version (I'm not an apple user) and get a dev cert for apple (to release it in the store, otherwise Apple Sillicon users won't be able to install it easily AFAIK).

In other words, in a month or so I'll fix it with the new 1.0.0 release.

If you can test Gitnuro and report the bugs that you find (or features request) it will help me to release the stable version for MacOS faster.

**Comment by JetpackDuba** [comment]
> In regards to apple silicone users they can install applications the same way that mac has been able to for decades.

If I'm not mistaken there are some [implications that may affect the UX](https://eclecticlight.co/2021/01/26/when-you-dont-have-permission-to-run-an-app-on-an-m1-mac/). 

> An apple dev cert is not necessary (unless you really want to publish in app store), a popular alternative for installing applications is Homebrew https://brew.sh/

I didn't know about Homebrew, looks like an interesting option.

In any case, I may provide multiple options, I'd rather not have to pay for the dev certificate but if it's necessary I will probably do it.

**Comment by ProjectInfinity** [comment]
https://github.com/JetpackDuba/Gitnuro/pull/13 is related to this issue.

**Comment by JetpackDuba** [comment]
v1.0.1 includes a MacOS release as a bundle & as a hombrew package.

Everything is unsigned for now. I have to consider if I am really willing to pay 100$/year to apple to sign it.
